packages/types/src/swap.ts

```diff
@@ -100,3 +113,59 @@ export const getOneWaySwapValues = (
     unfilled: getUnfilledAmount(swapView),
   };
 };
+
+/**
+ * Returns an object describing a swap: its input, its output, and any unfilled
+ * amount. This function is useful for displaying a swap in a UI, and it assumes
+ * we're dealing with a one-way swap. If passed a two-way swap, it will throw.
+ */
+export const getOneWaySwapValuesOpaque = (
+  swapView: SwapView,
+): {
+  input: ValueView;
+  output: ValueView;
+} => {
+  if (!isOneWaySwapOpaque(swapView)) {
```

Comment:
Ah here's the guard. I think this gives you license to use generic getters below that work for both opaque and visible cases.